plugins/module_utils/utils.py

```diff
@@ -31,12 +31,24 @@ def sync(state, client, path, payload, check_mode):
     return False, remote_object
 
 
+def is_different_type(a, b):
+    if isinstance(a, string_types) and isinstance(b, string_types):
+        return False
+    return (a and b) and not isinstance(a, type(b))
```

Comment:
Why do we need that `(a and b)` part?

Reply:
I only check type for non None types, because while it is a different type, it's not really a structural mismatch, it just means that that value is empty, which happens quite often when reading data form remote.

Reply:
I see, can you do it explicitly then, please, so it'll read better:

```python
if a is None or b is None:
  return False
else:
  return not isinstance(a, type(b))
```

Reply:
I renamed the function to `is_same_type`. I think it's a bit easier to read and we lose all the negating in the function.